Sources/Sentry/SentryScope.m

```diff
@@ -319,6 +326,33 @@ - (void)addAttachment:(SentryAttachment *)attachment
     }
 }
 
+- (void)setTransaction:(id<SentrySpan>)transaction
+{
+    self.currentTransaction = transaction;
+    [self notifyListeners];
+}
+
+- (void)setTransactionName:(NSString *)transactionName
+{
+    if (self.currentTransaction != nil &&
+        [self.currentTransaction isKindOfClass:[SentryTracer class]]) {
+        [(SentryTracer *)self.currentTransaction setName:transactionName];
+    }
+    self.fallbackTransactionName = transactionName;
+    [self notifyListeners];
```

Comment:
This would notify since this transaction field is part of the protocol and goes into crash events too.

Reply:
guess same as https://github.com/getsentry/sentry-cocoa/pull/992/files#r600487333